### PR TITLE
Feat/timer/UI

### PR DESCRIPTION
**Pull request by taghizadlaura**
# Implement TimerScreen's UI

## Description
This PR introduces the initial implementation of the `TimerScreen` UI to align with the provided Figma design. The task focuses on the visual representation of the timer, ensuring accurate padding, typography, and color application. Timer logic is not included in this scope.

It closes issue #210.

## Changes

- Developed the `TimerScreen` UI to match Figma specifications.
- Ensured responsive design and adherence to Material design guidelines.

#### Modified
- `TimerScreen.kt`: Implementation of the UI for the `TimerScreen`.
- `TimerScreenTest.kt` and `C.kt`: Added a few tests using testTags only.
- Updated `Dimens.kt `for dimensions adjustments.

## Dependencies Added
- None

## Testing
- Manual visual inspection to confirm consistency with the Figma design.
- Initial setup for UI tests with placeholders for functional validations once the timer logic is implemented.

## Screenshots

TimerScreen without time passing :

TimerScreen with time passing :

### CODE REVIEW

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 1]
```diff
@@ -22,9 +50,43 @@ import com.android.periodpals.ui.theme.dimens
 
 private const val SCREEN_TITLE = "Tampon Timer"
 
-/** TODO: Placeholder Screen, waiting for implementation */
+// Displayed text
+private const val DISPLAYED_TEXT_ONE =
+    "Start your tampon timer.\n" + "You’ll be reminded to change it !"
+private const val DISPLAYED_TEXT_TWO =
+    "You’ve got this. Stay strong !!\n" + "Don’t forget to stay hydrated !"
```
Comment (agonzalez-r) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:57]: nitpick: i know you followed exactly the text on the figma, but it looks werid to have "!!". Please remove the second exclamation point

File: app/src/test/java/com/android/periodpals/ui/timer/TimerScreenTest.kt [hunk 2]
```diff
@@ -42,6 +42,11 @@ class TimerScreenTest {
...
 
-    composeTestRule.onNodeWithTag(TimerScreen.TIMER_TEXT).performScrollTo().assertIsDisplayed()
+    // .performScrollTo().
+    composeTestRule.onNodeWithTag(TimerScreen.DISPLAYED_TEXT).performScrollTo().assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TimerScreen.CIRCULAR_PROGRESS_INDICATOR).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TimerScreen.HOURGLASS).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TimerScreen.START_STOP_BUTTON).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TimerScreen.USEFUL_TIP).assertIsDisplayed()
   }
 }
```
Comment (agonzalez-r) [inline app/src/test/java/com/android/periodpals/ui/timer/TimerScreenTest.kt:52]: You should add a test that ensures the functionality of the button: 
- assert that when it hasn't been pressed, the text of the button equals `Start`
- assert that when it has been pressed, the text equals `Stop`
- you can also assert any other changes triggered by the button (i.e the displayed text)

These tests will be completed when the Timer VM is integrated, but it's a good base!

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 3]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
     ) {
-      // TODO: delete when implementing the screen
-      Text("Timer Screen", modifier = Modifier.fillMaxSize().testTag(TimerScreen.TIMER_TEXT))
+
+      // Displayed text
+      Text(
+          text = correct_displayedText(isTimerRunning),
+          modifier = Modifier.testTag(TimerScreen.DISPLAYED_TEXT),
+          textAlign = TextAlign.Center,
+          style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:121]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 4]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+                    containerColor =
+                        if (isTimerRunning) MaterialTheme.colorScheme.error
+                        else MaterialTheme.colorScheme.primary,
+                    contentColor =
+                        if (isTimerRunning) MaterialTheme.colorScheme.onError
+                        else MaterialTheme.colorScheme.onPrimary)) {
+              Text(
+                  text = if (isTimerRunning) "STOP" else "START",
+                  textAlign = TextAlign.Center,
+                  style = MaterialTheme.typography.bodyLarge.copy(fontWeight = FontWeight.Bold))
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:153]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 5]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+            trackColor = MaterialTheme.colorScheme.primaryContainer,
+            strokeCap = StrokeCap.Round,
+        )
+
+        // Time displayed
+        Text(
+            text = formatedTime(timeLeft),
+            textAlign = TextAlign.Center,
+            style =
+                MaterialTheme.typography.bodyLarge.copy(
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:248]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 6]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+        Icon(
+            imageVector = Icons.Filled.Lightbulb,
+            contentDescription = "Lightbulb Icon",
+            tint = MaterialTheme.colorScheme.primary,
+            modifier = Modifier.size(MaterialTheme.dimens.iconSize),
+        )
+        Text(
+            text = "Useful Tip",
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.Bold),
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:337]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 7]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+        )
+      }
+
+  HorizontalDivider(
+      thickness = MaterialTheme.dimens.borderLine,
+      color = MaterialTheme.colorScheme.secondary.copy(alpha = 0.5f))
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:343]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 8]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+
+  Text(
+      text = usefulTipText,
+      textAlign = TextAlign.Center,
+      style = MaterialTheme.typography.bodyMedium,
+  )
+
+  HorizontalDivider(
+      thickness = MaterialTheme.dimens.borderLine,
+      color = MaterialTheme.colorScheme.secondary.copy(alpha = 0.5f))
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:353]: delete copy

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 9]
```diff
@@ -48,8 +112,243 @@ fun TimerScreen(navigationActions: NavigationActions) {
...
+                        if (isTimerRunning) MaterialTheme.colorScheme.onError
+                        else MaterialTheme.colorScheme.onPrimary)) {
+              Text(
+                  text = if (isTimerRunning) "STOP" else "START",
+                  textAlign = TextAlign.Center,
+                  style = MaterialTheme.typography.bodyLarge.copy(fontWeight = FontWeight.Bold))
+            }
+      }
+
+      Spacer(Modifier.size(MaterialTheme.dimens.small1))
```
Comment (Harrish92) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:157]: remove unnecessary spacer

**Review by Harrish92 — APPROVED** [review]
## PR Review

### Summary
Great Job on implementing the timer. There are still some changes that are needed.

### Important

#### Code Quality
- The code is generally well-structured and readable.
- Consistency in naming conventions and formatting is maintained

### Testing
- The tests cover most of the new code now. Good job.

### Documentation
- Comments and documentation are clear and helpful.

### Steps before PR approved
#### See the comments
- Remove the `Spacer` component, since the spacing is supposed to be regular. As you can see in this screenshot :
photo_2024-11-21_16-17-50
- Refactor the code to use theme, style, and padding values directly without copying them.

**Review by agonzalez-r — APPROVED** [review]
## Summary
Great job with the UI! the hourglass animation is really cool!
## Important 
#### Code Quality
LGTM
The code is clean and consistent, it follows the padding, colour and typography guidelines. It is also correctly modularised. 
#### Functionality
LGTM. All test pasts locally. 
## Testing
With the added tests for the button, I believe that for now it is all covered (90% coverage!).
## Documentation
All public and private functions are well documented. The TODO's will facilitate the integration of the VM. 
LGTM

**Pull request merged**